presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -126,4 +126,14 @@ public static String arraySortDesc()
     {
         return "RETURN reverse(array_sort(remove_nulls(input))) || filter(input, x -> x is null)";
     }
+
+    @SqlInvokedScalarFunction(value = "remove_nulls", deterministic = true, calledOnNullInput = true)
+    @Description("Removes null values from an array.")
+    @TypeParameter("T")
+    @SqlParameter(name = "input", type = "array(T)")
+    @SqlType("array<T>")
+    public static String removeNulls()
+    {
+        return "RETURN filter(input, x -> x is not null)";
```

Comment:
Let's optimize it for common case where there are usually no nulls:

RETURN IF(find_first(input, x->x IS NULL) IS NULL, x, filter(input, x -> x is not null))

Reply:
This won't work in case input contains nulls. I'm planning to use none_match instead.